packer/provisioner.go

```diff
@@ -123,10 +133,41 @@ func (h *ProvisionHook) Run(ctx context.Context, name string, ui Ui, comm Commun
 				"`communicator` config was set to \"none\". If you have any provisioners\n" +
 				"then a communicator is required. Please fix this to continue.")
 	}
+
+	cast := CastDataToMap(data)
+
+	if h.HCL2Prepare != nil {
+		// For HCL2, decode and prepare Provisioners now to interpolate build variables
+		coreP, diags := h.HCL2Prepare(h.ParentBuildName, h.ParentBuildType, cast)
+		if diags.HasErrors() {
+			return diags
+		}
+		for i, p := range coreP {
+			var pConfig interface{}
+			if len(p.config) > 0 {
+				pConfig = p.config[0]
+			}
+
+			if typeName(h.Provisioners[i].Provisioner) == typeName(DebuggedProvisioner{}) {
+				h.Provisioners[i] = &HookedProvisioner{
+					Provisioner: &DebuggedProvisioner{Provisioner: p.Provisioner},
+					Config:      pConfig,
+					TypeName:    p.PType,
+				}
+				continue
+			}
```

Comment:
Could `HCL2Prepare` do this ?

Reply:
You mean make `HCL2Prepare` return the `[]*HookedProvisioner` list and deal with the DebuggedProvisioner itself?

Reply:
yes

Reply:
I mean simply returning provisioners that have a `DebuggedProvisioner`.

Reply:
The provisioner is wrapped in a `DebuggedProvisioner` in the `CoreBuild.Run` method. I can make change the hcl.PackerConfig to keep the debug flag information and do the wrap together with the other provisioners' wrappers if is that what you mean. 🤔  This is the only way to add this wrap inside of HCL2Prepare without getting this information from the current hook.

Reply:
I did the option with the debug flag in the PackerConfig and I think it looks good! 👍

Reply:
I agree, super nice !